---
title: "102-redone-seq"
author: "Mac Campbell"
date: "2022-12-08"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
```

## Align and examine new data     
LFS1_S113_L006 LFS2_S114_L006
```{r}
meta<-read_csv("meta/NightSmLFS_Redband_sampledata.csv") %>% filter(Species %in% c("LarvalLFS","NightSmelt","LongfinSmelt")) %>% 
  mutate(Prefix=ifelse(Plate=="NightSmelt_LFS_Normalized_1","LFS1_S113_L006","LFS2_S114_L006"))

meta
```

batch rename, move to data/new-data     

```{r}
commands<-meta %>% mutate(Command1=paste0("mv /home/maccamp/missouri-trout/data/",Plate,"/",
                                          Prefix,"_RA_","GG",
                                          `WellBarcode`,"TGCAGG.fastq"," data/new-data/",
                                          `SampleID`,"_RA.fastq")) %>% 
  mutate(Command2=paste0("mv /home/maccamp/missouri-trout/data/",Plate,"/",
                         Prefix,"_RB_","GG",`WellBarcode`,"TGCAGG.fastq",
                                   " data/new-data/",`SampleID`,"_RB.fastq"))

c1<- select(commands, Command1) %>% rename(Command=Command1)
c2<- select(commands, Command2) %>% rename(Command=Command2)
c3 <- bind_rows(c1, c2)
write_tsv(c3, file="102.1-batch-rename.sh", col_names = FALSE)

```


```{sh, eval=FALSE}
# in split2
ls | grep RA | perl -pe 's/.fastq//g' > forward
ls | grep RB | perl -pe 's/.fastq//g' > reverse
ls | grep RA | perl -pe 's/_RA.fastq//g' > name
paste forward reverse name  > samples.txt
bash $HOME/missouri-trout/doAlign-unzipped.sh samples.txt /home/maccamp/genomes/hypomesus-20210204/Hyp_tra_F_20210204.fa 
```


Aligning 114 samples.......       

```{r}
existing<-read_tsv("meta/counts.tsv") %>% filter(Alignment !="Original")
existing<-existing %>% filter(Alignment=="New Genome") %>% filter(Reads > 1.5e5) %>%
  mutate(Path=paste0("/home/maccamp/data/longfin/",File))

ggplot(existing) + geom_histogram(aes(x=Reads))
```

```{r}
new<-read_csv("outputs/102/newseq.dat", col_names = c("Sample","Sort","Dedup"))

ggplot(new) + geom_histogram(aes(x=Dedup))
```
```{r}
together<-existing %>% select(-Alignment) %>% separate(Sample, into=c("Location","Number"), sep="_", remove=FALSE) %>% select(-Number)
new2<-new %>% mutate(File=paste0(Sample,".sort.flt.bam")) %>% rename(Reads=Dedup) %>% select(-Sort) %>%
  mutate(Path=paste0("data/new-data/",File)) %>% filter(Reads > 50000) %>%
  mutate(Location="New")

together<-together %>% bind_rows(new2)

write_tsv(select(together,Path), file="bamlists/188.bamlist", col_names = FALSE)

ggplot(together) +
  geom_histogram(aes(x=Reads, fill=Location))
```

```{sh, eval=FALSE}
srun -p high -t 14:00:00 --mem=16G --nodes=1 $HOME/angsd/angsd -P 24 \
  -ref /home/maccamp/genomes/hypomesus-20210204/Hyp_tra_F_20210204.fa \
  -bam $HOME/longfin/bamlists/188.bamlist -GL 1 \
  -doGLF 2 -doMajorMinor 1 -doMaf 2 -SNP_pval 1e-3 -minMapQ 10 -minQ 20 -minMaf 0.05 \
  -rf $HOME/delta-smelt/metadata/large-contigs.txt  -minInd 160 \
  -out $HOME/longfin/outputs/102/188-min160 > outputs/102/beagle.out 2> outputs/102/beagle.err &
  
srun -p high -t 01:00:00 --nodes=1 python $HOME/pcangsd/pcangsd.py -beagle outputs/102/188-min160.beagle.gz \
   -admix -o outputs/102/188-pca -threads 10

#found bestk=4 @ minind 141
#best k =7 @ 150 inds
#best k=5 @160 inds
```

25028 sites, too much missing data with minind 141
trying minind 166, 911 sites
50 inds 137815 sites?
150 inds 	-> Number of sites retained after filtering: 8199    
160 inds 1858 sites        
```{r}
cov<-read_delim("outputs/102/188-pca.cov", col_names=FALSE, delim=" ") %>% as.matrix()
meta<-together

meta$Location<-factor(meta$Location, levels=c("YBAK","SKNA","New","PTLC","HRLC","FRAS","LWSH","COLR",
                                              "HUMB","SFBY","PETA","ALVS","SUIB","CHPI"))
```

```{r}
#' @param samples character vector with the individuals IDs in the order in which
#' they were passed in the bamlist to angsd.
#' @param cov covariance matrix
covar2pcs <- function(samples, cov) {
  
  
  eig <- eigen(cov, symm = TRUE)
  PC <- as.data.frame(eig$vectors) %>%
    as_tibble() %>%
    setNames(sprintf("PC-%02d", 1:ncol(.)))
  
  samtib <- tibble(sample = samples)
  
  list(
    PCs = bind_cols(samtib, PC),
    eigevalues = eig$values
  )
}
```

```{r}
pca <- covar2pcs(meta$Sample, cov)

pca_long <- pca$PCs %>%
  tidyr::gather(., key = "PC", "val", -sample)

# then expand a grid of the possible comparisons (ordered)
expg <- expand.grid(sample = pca$PCs$sample,
                    PCx = sprintf("PC-%02d", 1:6),
                    PCy = sprintf("PC-%02d", 1:6),
                    stringsAsFactors = FALSE) %>%
  tibble::as_tibble()

# then left join the pca results onto that
pca_pairs <- dplyr::left_join(expg, pca_long, by = c("sample", "PCx" = "PC")) %>%
  dplyr::rename(val_x = val) %>%
  dplyr::left_join(pca_long, by = c("sample", "PCy" = "PC")) %>%
  dplyr::rename(val_y = val)

pp_meta <- pca_pairs %>%   # just keep the first 6 PCs around
  left_join(., meta, by = c("sample" = "Sample")) %>%
  mutate(group = Location) 
```

Plot    

```{r}
npc <- 3
pp_meta2 <- pp_meta %>%
  filter( (PCx %in% sprintf("PC-%02d", 1:npc)) & 
            (PCy %in% sprintf("PC-%02d", 1:npc)) )

eig <- eigen(cov, symm = TRUE)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```

```{r}
ggplot(pp_meta2, aes(x = val_x, y = val_y, color=Location)) +
  geom_point() +
  facet_grid(PCx ~ PCy)
```

```{r}
sub12<-pp_meta2 %>% filter( (PCx =="PC-01") & (PCy =="PC-02") )
text<- sub12 %>% select(sample,Location,val_x,val_y) %>% group_by(Location) %>%
  mutate(meanX=mean(val_x), meanY=mean(val_y)) %>% select(Location, meanX, meanY) %>%
  unique()   

ggplot(sub12, aes(x = val_x, y = val_y, color=Location))+
  geom_point(size = 2, alpha=0.5) +
  geom_text_repel(data=text, aes(x=meanX, y=meanY, label=Location), max.overlaps = Inf) +
  scale_fill_discrete(na.value = "white") + 
  theme_bw()+
  theme(panel.grid=element_blank())+
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = ""))+
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = ""))+
  theme(plot.title = element_text(hjust = 0.5)) +
  scale_color_viridis_d(option="C")

ggsave("outputs/102/lfs-dsm-reference-pc12.jpeg")
```

Ohh gross. Maybe I'll need to downsample again?


```{r}
together %>% filter(Location == "New") %>% summarize(Mean=mean(Reads))
together %>% filter(Location != "New") %>% summarize(Mean=mean(Reads))
```

Downsample to 150K?    

```{r}
downsample<-read_tsv("meta/counts.tsv") %>% filter(Alignment !="Original") %>% filter(Reads > 1.5e5) %>%
  mutate(Frac=1.5e5/Reads) %>%
  mutate(ReductionCommand = paste0("samtools view -bs ",Frac, " ", "/home/maccamp/data/longfin/",
                                   Sample, "_R1.sort.flt.bam"," > ",
                                   "/home/maccamp/longfin/data/reduced/",
                                   Sample,".reduced.bam" )) %>%
  mutate(newPath= paste0("data/reduced/",Sample,".reduced.bam" ))

write_csv(downsample$ReductionCommand %>% as_tibble(), "102.1-downsample.sh", col_names = FALSE)
```


```{r}
lfs<-downsample %>% select(-Alignment,-ReductionCommand) %>%
  separate(Sample, into=c("Location","Number"), sep="_", remove=FALSE) %>% select(-Number) %>%
  rename(Path=newPath)


lfs<-lfs %>% bind_rows(new2%>% mutate(Frac=1))

write_tsv(select(lfs,Path), file="bamlists/188-sub.bamlist", col_names = FALSE)

ggplot(lfs) +
  geom_histogram(aes(x=(Frac*Reads), fill=Location))
```

```{sh, eval=FALSE}
srun -p high -t 14:00:00 --mem=16G --nodes=1 $HOME/angsd/angsd -P 24 \
  -ref /home/maccamp/genomes/hypomesus-20210204/Hyp_tra_F_20210204.fa \
  -bam $HOME/longfin/bamlists/sub-188.bamlist -GL 1 \
  -doGLF 2 -doMajorMinor 1 -doMaf 2 -SNP_pval 1e-3 -minMapQ 10 -minQ 20 -minMaf 0.05 \
  -rf $HOME/delta-smelt/metadata/large-contigs.txt  -minInd 141 \
  -out $HOME/longfin/outputs/102/188-sub> outputs/102/beagle.out 2> outputs/102/beagle.err &
  
srun -p high -t 01:00:00 --nodes=1 python $HOME/pcangsd/pcangsd.py -beagle outputs/102/188-sub.beagle.gz \
   -admix -o outputs/102/188-sub-pca -threads 10
```
