---
title: "102-redone-seq"
author: "Mac Campbell"
date: "2022-12-08"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
```

## Align and examine new data     
LFS1_S113_L006 LFS2_S114_L006
```{r}
meta<-read_csv("meta/NightSmLFS_Redband_sampledata.csv") %>% filter(Species %in% c("LarvalLFS","NightSmelt","LongfinSmelt")) %>% 
  mutate(Prefix=ifelse(Plate=="NightSmelt_LFS_Normalized_1","LFS1_S113_L006","LFS2_S114_L006"))

meta
```

batch rename, move to data/new-data     

```{r}
commands<-meta %>% mutate(Command1=paste0("mv /home/maccamp/missouri-trout/data/",Plate,"/",
                                          Prefix,"_RA_","GG",
                                          `WellBarcode`,"TGCAGG.fastq"," data/new-data/",
                                          `SampleID`,"_RA.fastq")) %>% 
  mutate(Command2=paste0("mv /home/maccamp/missouri-trout/data/",Plate,"/",
                         Prefix,"_RB_","GG",`WellBarcode`,"TGCAGG.fastq",
                                   " data/new-data/",`SampleID`,"_RB.fastq"))

c1<- select(commands, Command1) %>% rename(Command=Command1)
c2<- select(commands, Command2) %>% rename(Command=Command2)
c3 <- bind_rows(c1, c2)
write_tsv(c3, file="102.1-batch-rename.sh", col_names = FALSE)

```


```{sh, eval=FALSE}
# in split2
ls | grep RA | perl -pe 's/.fastq//g' > forward
ls | grep RB | perl -pe 's/.fastq//g' > reverse
ls | grep RA | perl -pe 's/_RA.fastq//g' > name
paste forward reverse name  > samples.txt
bash $HOME/missouri-trout/doAlign-unzipped.sh samples.txt /home/maccamp/genomes/hypomesus-20210204/Hyp_tra_F_20210204.fa 
```


Aligning 114 samples.......       

```{r}
existing<-read_tsv("meta/counts.tsv") %>% filter(Alignment !="Original")
existing<-existing %>% filter(Alignment=="New Genome") %>% filter(Reads > 1.5e5) %>%
  mutate(Path=paste0("/home/maccamp/data/longfin/",File))

ggplot(existing) + geom_histogram(aes(x=Reads))
```

```{r}
new<-read_csv("outputs/102/newseq.dat", col_names = c("Sample","Sort","Dedup"))

ggplot(new) + geom_histogram(aes(x=Dedup))
```
```{r}
together<-existing %>% select(-Alignment)
new2<-new %>% mutate(File=paste0(Sample,".sort.flt.bam")) %>% rename(Reads=Dedup) %>% select(-Sort) %>%
  mutate(Path=paste0("data/new-data/",File)) %>% filter(Reads > 50000)

together<-together %>% bind_rows(new2)

write_tsv(select(together,Path), file="bamlists/188.bamlist", col_names = FALSE)
```

```{sh, eval=FALSE}
srun -p high -t 14:00:00 --mem=16G --nodes=1 $HOME/angsd/angsd -P 24 \
  -ref /home/maccamp/genomes/hypomesus-20210204/Hyp_tra_F_20210204.fa \
  -bam $HOME/longfin/bamlists/188.bamlist -GL 1 \
  -doGLF 2 -doMajorMinor 1 -doMaf 2 -SNP_pval 1e-6 -minMapQ 10 -minQ 20 -minMaf 0.05 \
  -rf $HOME/delta-smelt/metadata/large-contigs.txt  -minInd 141 \
  -out $HOME/longfin/outputs/102/188 > outputs/102/beagle.out 2> outputs/102/beagle.err &
```
