---
title: "102-demulti-new"
author: "Mac Campbell"
date: "8/22/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
```

## Demulti new samples

Counting lines in the files to make sure demulti was complete. For example:
 awk '{l++} END {print l/4}' LibPlate25_MORedband02_R1.fastq
 
 l004:
gunzip -c BMAG072_S1_L004_R1_001.fastq.gz |  awk '{l++} END {print l/4}'
cat *R1.fastq | awk '{l++} END {print l/4}' 
These are 


We have three plates in two lanes. Due to space constraints, they are in two locations. 

Checking completeness:
`(base) maccamp@farm:/group/millermrgrp4/maccamp/alanas-trout-2$  gunzip -c BMAG072_S1_L004_R1_001.fastq.gz |  awk '{l++} END {print l/4}'`
2947604505    

__1__ Combine into one file     

Doing in data/combined-plates   

```{sh, eval=FALSE}
cat /group/millermrgrp4/maccamp/alanas-trout/LibPlate21_LongfinSmelt01_R1.fastq          /group/millermrgrp4/maccamp/alanas-trout-2/LibPlate21_LongfinSmelt01_R1.fastq > longfin-01_R1.fastq &

cat /group/millermrgrp4/maccamp/alanas-trout/LibPlate22_LongfinSmelt02_R1.fastq          /group/millermrgrp4/maccamp/alanas-trout-2/LibPlate22_LongfinSmelt02_R1.fastq > longfin-02_R1.fastq &

cat /group/millermrgrp4/maccamp/alanas-trout/LibPlate23_LongfinSmelt03_R1.fastq          /group/millermrgrp4/maccamp/alanas-trout-2/LibPlate23_LongfinSmelt03_R1.fastq > longfin-03_R1.fastq &


cat /group/millermrgrp4/maccamp/alanas-trout/LibPlate21_LongfinSmelt01_R2.fastq          /group/millermrgrp4/maccamp/alanas-trout-2/LibPlate21_LongfinSmelt01_R2.fastq > longfin-01_R2.fastq &

cat /group/millermrgrp4/maccamp/alanas-trout/LibPlate22_LongfinSmelt02_R2.fastq          /group/millermrgrp4/maccamp/alanas-trout-2/LibPlate22_LongfinSmelt02_R2.fastq > longfin-02_R2.fastq &

cat /group/millermrgrp4/maccamp/alanas-trout/LibPlate23_LongfinSmelt03_R2.fastq          /group/millermrgrp4/maccamp/alanas-trout-2/LibPlate23_LongfinSmelt03_R2.fastq > longfin-03_R2.fastq &
```

__2__ demultiplex each plate
      Checking for barcodes
      grep GGACAAGCTATGCAGG --color longfin-03_R1.fastq (works)
      barcode is ACAAGCTA
      
      separate dirs in data/   
      Using column 2 identifier and column 9    
      
```{sh, eval=FALSE}
    cat ../../meta/sample-meta.tsv | grep "Plate 1" | awk '{ print " perl ./BarcodeSplitListBestRadPairedEnd.pl", "longfin-01_R1.fastq", "longfin-01_R2.fastq", "GG"$9"TGCAGG", $2 }' > tasks.sh  
             srun -p high -t 10:00:00 --nodes=1 parallel -j 24 < tasks.sh > std.out 2>std.err &   

      cat ../../meta/sample-meta.tsv | grep "Plate 2" | awk '{ print " perl ./BarcodeSplitListBestRadPairedEnd.pl", "longfin-02_R1.fastq", "longfin-02_R2.fastq", "GG"$9"TGCAGG", $2 }' > tasks.sh  
           srun -p high -t 10:00:00 --nodes=1 parallel -j 24 < tasks.sh > std.out 2>std.err &   

     
      cat ../../meta/sample-meta.tsv | grep "Plate 3" | awk '{ print " perl ./BarcodeSplitListBestRadPairedEnd.pl", "longfin-03_R1.fastq", "longfin-03_R2.fastq", "GG"$9"TGCAGG", $2 }' > tasks.sh  
      srun -p high -t 10:00:00 --nodes=1 parallel -j 24 < tasks.sh > std.out 2>std.err &   
```
      
      
## Alignment

Previously, I used the non-accessioned genome.

`(base) maccamp@farm:~/data/longfin$ samtools view -H ALVS_097_R1.sort.bam`
`@HD	VN:1.5	SO:coordinate`
`@SQ	SN:lg01	LN:12516727`
`@SQ	SN:lg02	LN:17206656`
`@SQ	SN:lg03	LN:15692960`
`bwa mem /home/maccamp/genomes/hypomesus-20210204/Hyp_tra_F_20210204.fa ALVS_097_R1.fastq.gz ALVS_097_R2.fastq.gz`

