---
title: "105-integrating-data"
author: "Mac Campbell"
date: "2023-01-06"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
library(ggrepel)
library(ggpubr)
```

1. Really good to know Russian River LF sample #'s 22-0074 22-0102, only two fish total     

## Existing data

```{r}
existing<-read_tsv("meta/counts.tsv") %>% filter(Alignment !="Original")
existing<-existing %>% filter(Alignment=="New Genome") %>%
  mutate(Path=paste0("/home/maccamp/data/longfin/",File)) %>% filter(Reads > 1.5e5)

ggplot(existing) + geom_histogram(aes(x=Reads))
```

```{r}
existing %>% summarize(Mean=mean(Reads), Median=median(Reads), max(Reads))
  
```

## New Data

```{r}
dat<-read_csv("outputs/104/alt.dat", col_names = c("Sample","Sort","Dedup"))

ds<-dat %>% filter(Dedup > 1.5e5) %>% mutate(Path=paste0("data/alt-bams/",Sample,".sort.flt.bam"))

ggplot(ds) +
  geom_histogram(aes(x=Dedup))
```

```{r}
mm<-existing %>% select(-File) %>% rename(Dedup=Reads) %>% select(-Alignment) %>% mutate(Origin="Old")
mm %>% bind_rows(ds %>% select(-Sort) %>% mutate(Origin="New"))
```

## Combine by trimming then downsampling

```{r}
plate1<-dat[startsWith(dat$Sample, "HS2253"),] 
plate1<-plate1 %>% mutate(Plate="plate-1")
plate1$Barcode<-gsub("HS2253_AGTCAA-","",plate1$Sample)

plate23<-dat[(!startsWith(dat$Sample, "HS2253")),] 
plate23<-plate23 %>% separate(Sample, into = c("p1","p2","Barcode"), remove=FALSE) %>% mutate(Plate=paste0(p1,"-",p2)) %>% select(-p1,-p2)

plates<-bind_rows(plate1, plate23)

#HS2253_AGTCAA_RA_GGTGGCTTCATGCAGG.fastq

newseqs<- plates %>% select(-Sample) %>% mutate(Sample=paste0(Plate,"-",Barcode)) %>%
  mutate(PathRA = ifelse(Plate=="plate-1", paste0("data/alt-1/", "HS2253_AGTCAA_RA_GG", Barcode,"TGCAGG.fastq"),
                  ifelse(Plate=="plate-2", paste0("data/alt-2/", "HS2253_AGTCAA_RA_GG", Barcode,"TGCAGG.fastq"),
                  ifelse(Plate=="plate-3", paste0("data/alt-3/", "HS2253_AGTCAA_RA_GG", Barcode,"TGCAGG.fastq"),"NULL")))) %>%
  mutate(PathRB = ifelse(Plate=="plate-1", paste0("data/alt-1/", "HS2253_AGTCAA_RB_GG", Barcode,"TGCAGG.fastq"),
                  ifelse(Plate=="plate-2", paste0("data/alt-2/", "HS2253_AGTCAA_RB_GG", Barcode,"TGCAGG.fastq"),
                  ifelse(Plate=="plate-3", paste0("data/alt-3/", "HS2253_AGTCAA_RB_GG", Barcode,"TGCAGG.fastq"),"NULL")))) 
```


My forward are 135 bp in length theirs are 85, trimming 50 bp
My reverse are 151 bp in length, theirs are 101, trimming 50 bp

'module load seqtk'    

```{r}
rename<-newseqs %>% mutate(r1=paste0("seqtk trimfq -e 50 ", PathRA, " > data/alt-trim/", Sample,"_RA.fastq")) %>% 
                    mutate(r2=paste0("seqtk trimfq -e 50 ", PathRB, " > data/alt-trim/", Sample,"_RB.fastq"))
r1<- select(rename, r1) %>% rename(Command=r1)
r2<- select(rename, r2) %>% rename(Command=r2)
r3 <- bind_rows(r1, r2)
r3
write_tsv(r3, "105.1-batch-trim.sh",col_names = FALSE)
```

srun -p bigmemh -t 02:00:00 --nodes=1 parallel -j 10 < 105.1-batch-trim.sh > outputs/105/trim.stdout 2> outputs/105/trim.stderr 

Trim and re-align
```{sh, eval=FALSE}
ls | grep RA | perl -pe 's/.fastq//g' > forward
ls | grep RB | perl -pe 's/.fastq//g' > reverse
ls | grep RA | perl -pe 's/_RA.fastq//g' > name
paste forward reverse name  > samples.txt
bash $HOME/missouri-trout/doAlign-unzipped.sh samples.txt /home/maccamp/genomes/hypomesus-20210204/Hyp_tra_F_20210204.fa 
```

Then downsample!!     


```{r}
reads<-read_csv("outputs/105/alt-trim-reads.txt", col_names=c("Sample","Aligned","Dedup"))
m2<-rename %>% select(-Sort, -Dedup, -PathRA, -PathRB, -r1, -r2) %>% left_join(reads)
m2 %>% filter(Dedup > 1e5) %>% 
  summarize(Mean=mean(Aligned), Median=median(Dedup))
```

```{r}
ggplot(m2 %>% filter(Dedup > 1e5)) +
   geom_histogram(aes(x=Dedup))
```

Downsample to 1e6

```{r}
m3<-m2 %>% filter(Dedup > 1e5) %>% mutate(Frac=1e6/Dedup)  %>% 
  mutate(Path=ifelse(Dedup > 1e6, paste0("data/alt-trim/",Sample,".reduced.bam" ),
                paste0("data/alt-trim/",Sample,".sort.flt.bam")))

downsample<-m3 %>% filter(Dedup > 1e6 ) %>%
  mutate(ReductionCommand = paste0("samtools view -bs ",Frac, " ", "/home/maccamp/longfin/data/alt-trim/",
                                   Sample, ".sort.flt.bam"," > ",
                                   "/home/maccamp/longfin/data/alt-trim/",
                                   Sample,".reduced.bam" )) 

write_csv(downsample$ReductionCommand %>% as_tibble(), "105.1-downsample.sh", col_names = FALSE)


m3
```

srun -p high -t 02:00:00 --nodes=1 parallel -j 10 < 105.1-downsample.sh > outputs/105/downsample.stdout 2> outputs/105/downsample.stderr 

```{r}
m192<-m3 %>% filter(Dedup > 2e5)
write_tsv(m192 %>% select(Path), col_names = FALSE, "bamlists/192.bamlist")

names<-read_tsv("meta/non-lfs.tsv")
names$Sample<-gsub("HS2253_AGTCAA","plate-1",names$Sample)
m155<-m3 %>% filter(!(Sample %in% names$Sample)) 

write_tsv(m155 %>% select(Path), col_names = FALSE, "bamlists/155.bamlist")

```

m192
     
     
```{sh, eval=FALSE}
srun -p high -t 14:00:00 --mem=16G --nodes=1 $HOME/angsd/angsd -P 12 \
  -ref /home/maccamp/genomes/hypomesus-20210204/Hyp_tra_F_20210204.fa \
  -bam $HOME/longfin/bamlists/155.bamlist -GL 1 \
  -doGLF 2 -doMajorMinor 1 -doMaf 2 -SNP_pval 1e-6 -minMapQ 10 -minQ 20 -minMaf 0.05 \
  -r lg01 \
   -minInd 116 \
  -out $HOME/longfin/outputs/105/155-beagle > outputs/105/beagle-contigs.out 2> outputs/105/beagle-contigs.err &
```

-rf $HOME/delta-smelt/metadata/large-contigs.txt  
   
-r lg01 9420 sites with all species
-r lg01 2241 sites with LFS

```{sh, eval=FALSE}
python $HOME/pcangsd/pcangsd.py -beagle outputs/105/155-beagle.beagle.gz \
   -admix -o outputs/105/155-pca -threads 10
```




```{r}
#cov<-read_delim("outputs/105/155-pca.cov", col_names=FALSE, delim=" ") %>% as.matrix()
cov<-read_delim("outputs/105/192-pca.cov", col_names=FALSE, delim=" ") %>% as.matrix()

```

```{r}
#' @param samples character vector with the individuals IDs in the order in which
#' they were passed in the bamlist to angsd.
#' @param cov covariance matrix
covar2pcs <- function(samples, cov) {
  
  
  eig <- eigen(cov, symm = TRUE)
  PC <- as.data.frame(eig$vectors) %>%
    as_tibble() %>%
    setNames(sprintf("PC-%02d", 1:ncol(.)))
  
  samtib <- tibble(sample = samples)
  
  list(
    PCs = bind_cols(samtib, PC),
    eigevalues = eig$values
  )
}
```

```{r}
meta<-m192
pca <- covar2pcs(meta$Sample, cov)

pca_long <- pca$PCs %>%
  tidyr::gather(., key = "PC", "val", -sample)

# then expand a grid of the possible comparisons (ordered)
expg <- expand.grid(sample = pca$PCs$sample,
                    PCx = sprintf("PC-%02d", 1:6),
                    PCy = sprintf("PC-%02d", 1:6),
                    stringsAsFactors = FALSE) %>%
  tibble::as_tibble()

# then left join the pca results onto that
pca_pairs <- dplyr::left_join(expg, pca_long, by = c("sample", "PCx" = "PC")) %>%
  dplyr::rename(val_x = val) %>%
  dplyr::left_join(pca_long, by = c("sample", "PCy" = "PC")) %>%
  dplyr::rename(val_y = val)

pp_meta <- pca_pairs %>%   # just keep the first 6 PCs around
  left_join(., meta, by = c("sample" = "Sample")) 
```

Plot    

```{r}
npc <- 3
pp_meta2 <- pp_meta %>%
  filter( (PCx %in% sprintf("PC-%02d", 1:npc)) & 
            (PCy %in% sprintf("PC-%02d", 1:npc)) )

eig <- eigen(cov, symm = TRUE)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```


```{r}
ggplot(pp_meta2, aes(x = val_x, y = val_y)) +
  geom_point() +
  facet_grid(PCx ~ PCy)
```
Identify those wayward souls:
1] "SFBY_075"         "SFBY_083"         "plate-3-AAGACGGA" "plate-3-ACATTGGC" "plate-3-AGTCAAGC"
[6] "plate-3-TCCGTCTA"    
```{r}
subs<-pp_meta2 %>% filter(sample %in% c("plate-3-AAGACGGA","plate-3-ACATTGGC","plate-3-AGTCAAGC",
                                  "plate-3-TCCGTCTA"))

ggplot(pp_meta2 %>% left_join(names, by=c("sample"="Sample")), aes(x = val_x, y = val_y)) +
  geom_point(alpha=0.75, aes(color=Species)) + 
  geom_point(data=subs, aes(x=val_x, y=val_y),color="blue", alpha=.9) +
  facet_grid(PCx ~ PCy)
```
```{r}
sub12<-pp_meta2 %>% filter( (PCx =="PC-01") & (PCy =="PC-02") ) %>% left_join(names, by=c("sample"="Sample"))
subs12<-sub12 %>% filter(sample %in% c("plate-3-AAGACGGA","plate-3-ACATTGGC","plate-3-AGTCAAGC",
                                  "plate-3-TCCGTCTA"))
ggplot(sub12, aes(x = val_x, y = val_y)) +
  geom_point(alpha=0.75, aes(color=Species)) + 
  geom_point(data=subs12, aes(x=val_x, y=val_y),color="blue", alpha=.9) +
  xlim(0,0.05)
```

Some eulachon, a Hypomesus, a night smelt

## Merging

```{r}
e2<-existing %>% filter(Reads > 2e5)%>% select(-File) %>% 
  rename(Dedup=Reads) %>% select(-Alignment) %>% 
  separate(Sample, into = c("Location","Number"), remove=FALSE) %>% select(-Number)

m140<-m155 %>% filter(Dedup > 2e5) %>% select(Sample, Dedup, Path) %>% mutate(Location="New Sequencing")

c<-bind_rows(e2,m140)
write_tsv(select(c,Path), "bamlists/283.bamlist",col_names = FALSE)
#Dropping SFBY and some those wayward fish

c2<-c %>% filter(!Sample %in% c("plate-3-AAGACGGA","plate-3-ACATTGGC","plate-3-AGTCAAGC",
                                  "plate-3-TCCGTCTA")) %>% 
  filter(Location !="SFBY")
write_tsv(select(c2,Path), "bamlists/260.bamlist",col_names = FALSE)
meta260<-c2
```

```{sh, eval=FALSE}
srun -p high -t 5:00:00 --mem=16G --nodes=1 $HOME/angsd/angsd -P 12 \
  -ref /home/maccamp/genomes/hypomesus-20210204/Hyp_tra_F_20210204.fa \
  -bam $HOME/longfin/bamlists/260.bamlist -GL 1 \
  -doGLF 2 -doMajorMinor 1 -doMaf 2 -SNP_pval 1e-6 -minMapQ 10 -minQ 20 -minMaf 0.05 \
   -rf $HOME/delta-smelt/metadata/large-contigs.txt  \
   -minInd 195 \
  -out $HOME/longfin/outputs/105/260-beagle > outputs/105/beagle-contigs.out 2> outputs/105/beagle-contigs.err &
  

python $HOME/pcangsd/pcangsd.py -beagle outputs/105/260-beagle.beagle.gz \
  -o outputs/105/260-pca -threads 10
```

-r lg01 2323
all chroms: 76108


```{r}
cov<-read_delim("outputs/105/260-pca.cov", col_names=FALSE, delim=" ") %>% as.matrix()
```

```{r}
meta<-meta260
meta$Location<-factor(meta$Location, levels=c("YBAK","SKNA","PTLC","HRLC","FRAS","LWSH",
                                              "NOOK","COLR",
                                              "HUMB","PETA","ALVS","SUIB","CHPI",
                                         "New Sequencing"))

pca <- covar2pcs(meta$Sample, cov)

pca_long <- pca$PCs %>%
  tidyr::gather(., key = "PC", "val", -sample)

# then expand a grid of the possible comparisons (ordered)
expg <- expand.grid(sample = pca$PCs$sample,
                    PCx = sprintf("PC-%02d", 1:6),
                    PCy = sprintf("PC-%02d", 1:6),
                    stringsAsFactors = FALSE) %>%
  tibble::as_tibble()

# then left join the pca results onto that
pca_pairs <- dplyr::left_join(expg, pca_long, by = c("sample", "PCx" = "PC")) %>%
  dplyr::rename(val_x = val) %>%
  dplyr::left_join(pca_long, by = c("sample", "PCy" = "PC")) %>%
  dplyr::rename(val_y = val)

pp_meta <- pca_pairs %>%   # just keep the first 6 PCs around
  left_join(., meta, by = c("sample" = "Sample")) 
```


```{r}
npc <- 3
pp_meta2 <- pp_meta %>%
  filter( (PCx %in% sprintf("PC-%02d", 1:npc)) & 
            (PCy %in% sprintf("PC-%02d", 1:npc)) ) %>%
  mutate(Origin=ifelse(Location=="New Sequencing","New","Old"))

eig <- eigen(cov, symm = TRUE)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```

```{r}
ggplot(pp_meta2, aes(x = val_x, y = val_y)) +
  geom_point(aes(color=Location)) + 
  facet_grid(PCx ~ PCy)
```

```{r}
sub12<-pp_meta2 %>% filter( (PCx =="PC-01") & (PCy =="PC-02") )
text<- sub12 %>% select(sample,Location,val_x,val_y) %>% group_by(Location) %>%
  mutate(meanX=mean(val_x), meanY=mean(val_y)) %>% select(Location, meanX, meanY) %>%
  unique()   

pc12<-ggplot(sub12, aes(x = val_x, y = val_y, color=Location))+
  geom_point(size = 2, alpha=0.75) +
  scale_fill_discrete(na.value = "white") + 
  theme_bw()+
  theme(panel.grid=element_blank())+
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = ""))+
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = ""))+
  theme(legend.position = "")+
  theme(plot.title = element_text(hjust = 0.5)) +
  facet_wrap(.~Origin)

sub13<-pp_meta2 %>% filter( (PCx =="PC-01") & (PCy =="PC-03") )

pc13<-ggplot(sub13, aes(x = val_x, y = val_y, color = Location)) +
  geom_point(size = 2, alpha=0.75) +
  scale_fill_discrete(na.value = "white") + 
  theme_bw()+
  theme(panel.grid=element_blank())+
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = ""))+
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = ""))+
  theme(plot.title = element_text(hjust = 0.5)) +
  facet_wrap(.~Origin)

pcs<-ggarrange(pc12, pc13, ncol= 2, widths=c(1,1.5))
pcs
ggsave("outputs/105/combined-pca-260.jpeg", width=12)
```

```{r}
nooks<-sub12 %>% filter(Location=="New Sequencing") %>% filter(val_y < 0) %>% select(sample) %>%
  unique()

nooks$sample

dd<-pp_meta2 
dd$Location<-as.character(dd$Location)
dd<-dd %>% mutate(Loc=ifelse(sample %in% nooks$sample, "NOOK", Location))
dd$Location<-factor(dd$Loc, levels=c("YBAK","SKNA","PTLC","HRLC","FRAS","LWSH",
                                              "NOOK","COLR",
                                              "HUMB","PETA","ALVS","SUIB","CHPI",
                                         "New Sequencing"))

meta$Location<-as.character(meta$Location)
meta<-meta %>% mutate(Loc=ifelse(Sample %in% nooks$sample, "NOOK", Location))

meta$Location<-factor(meta$Loc, levels=c("YBAK","SKNA","PTLC","HRLC","FRAS","LWSH",
                                              "NOOK","COLR",
                                              "HUMB","PETA","ALVS","SUIB","CHPI",
                                         "New Sequencing"))

```

```{r}
ggplot(dd, aes(x = val_x, y = val_y)) +
  geom_point(aes(color=Location)) + 
  facet_grid(PCx ~ PCy)
```
```{r}
sub12<-dd %>% filter( (PCx =="PC-01") & (PCy =="PC-02") )
text<- sub12 %>% select(sample,Location,val_x,val_y) %>% group_by(Location) %>%
  mutate(meanX=mean(val_x), meanY=mean(val_y)) %>% select(Location, meanX, meanY) %>%
  unique()   

pc12<-ggplot(sub12, aes(x = val_x, y = val_y, color=Location))+
  geom_point(size = 2, alpha=0.75) +
  geom_label_repel(data=text, aes(x=meanX, y=meanY, color=Location,
                                  label=Location), max.overlaps = Inf) +
  scale_fill_discrete(na.value = "white") + 
  theme_bw()+
  theme(panel.grid=element_blank())+
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = ""))+
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = ""))+
  theme(legend.position = "")+
  theme(plot.title = element_text(hjust = 0.5))

sub13<-dd %>% filter( (PCx =="PC-01") & (PCy =="PC-03") )
text13 <- sub13 %>% select(sample,Location,val_x,val_y) %>% group_by(Location) %>%
  mutate(meanX=mean(val_x), meanY=mean(val_y)) %>% select(Location, meanX, meanY) %>%
  unique() 

pc13<-ggplot(sub13, aes(x = val_x, y = val_y, color = Location)) +
  geom_point(size = 2, alpha=0.75) +
  scale_fill_discrete(na.value = "white") + 
  geom_label_repel(data=text13, aes(x=meanX, y=meanY, color=Location,
                                  label=Location), max.overlaps = Inf) +
   theme_bw()+
  theme(panel.grid=element_blank())+
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = ""))+
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = ""))+
    theme(legend.position = "")+
  theme(plot.title = element_text(hjust = 0.5)) 

pcs<-ggarrange(pc12, pc13, ncol= 2, widths=c(1,1))
pcs
ggsave("outputs/105/combined-pca-260.jpeg", width=10)
```

     
```{sh, eval=FALSE}
srun -p high -t 01:00:00 --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/105/260-beagle.beagle.gz -K 2 -o outputs/105/260-K2
srun -p high -t 01:00:00 --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/105/260-beagle.beagle.gz -K 3 -o outputs/105/260-K3
srun -p high -t 01:00:00 --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/105/260-beagle.beagle.gz -K 4 -o outputs/105/260-K4
srun -p high -t 01:00:00 --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/105/260-beagle.beagle.gz -K 5 -o outputs/105/260-K5

```


```{r}
q2<-read_delim("outputs/105/260-K2.qopt", delim=" ", col_names=FALSE) %>% select(X1, X2) %>%
  rename(Q1=X1, Q2=X2)

df2<-bind_cols(q2, meta)            

q2s<-df2 %>% select(Location, Sample, Q1, Q2) %>% arrange(Location,Sample) %>%
  mutate(Index=1:n()) %>% gather(key=Ancestry, value=Q, 3:4) %>% mutate(K=2)

pops2<-q2s  %>% group_by(Location) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(Location,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2)) %>% ungroup() %>% unique()

```

```{r}
p2 <- ggplot(q2s) + 
  geom_col(aes(x=Index,y=Q, fill=Ancestry), color="NA", size = 0, width = 1) +
  geom_segment(data=pops2, x = pops2$Start - 0.5, y=0, xend = pops2$Start-0.5, yend=1, alpha=0.9, size=0.25) +
  geom_segment(data=pops2, x = pops2$Stop[length(pops2$Stop)]  + 
                 0.5, y=0, xend= pops2$Stop[length(pops2$Stop)] + 0.5, yend=1,  alpha=0.9, size=0.25) +
  geom_segment(x=0.5, xend= pops2$Stop[length(pops2$Stop)]+0.5, y=1, yend=1, alpha=0.9, size=0.25) +
  geom_segment(x=0.5, xend= pops2$Stop[length(pops2$Stop)]+0.5, y=0, yend=0, alpha=0.9, size=0.25) +
  ylim(-0.1,1.01) +
  xlim(-0.1, pops2$Stop[length(pops2$Stop)]+1) +
  theme(panel.background = element_blank()) +
  xlab("") + theme(legend.position = "") +
 # theme(axis.text.x = element_blank(), axis.ticks.x=element_blank()) +
  ggtitle("K = 2") +
  theme(plot.title = element_text(hjust = 0.5, face = "bold")) +
  scale_x_continuous(breaks=pops2$Position, labels=pops2$Location) +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  scale_fill_viridis_d(option="B")


p2

ggsave("outputs/105/lfs-dsm-reference-k2.jpg")
```

```{r}
q3<-read_delim("outputs/105/260-K3.qopt", delim=" ", col_names=FALSE) %>% select(X1, X2, X3) %>%
  rename(Q1=X1, Q2=X2, Q3=X3)

df3<-bind_cols(q3, meta)            

q3s<-df3 %>% select(Location, Sample, Q1, Q2, Q3) %>% arrange(Location,Sample) %>%
  mutate(Index=1:n()) %>% gather(key=Ancestry, value=Q, 3:5) %>% mutate(K=3)

pops3<-q3s  %>% group_by(Location) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(Location,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2)) %>% ungroup() %>% unique()

```

```{r}
p3 <- ggplot(q3s) + 
  geom_col(aes(x=Index,y=Q, fill=Ancestry), color="NA", size = 0, width = 1) +
  geom_segment(data=pops2, x = pops2$Start - 0.5, y=0, xend = pops2$Start-0.5, yend=1, alpha=0.9, size=0.25) +
  geom_segment(data=pops2, x = pops2$Stop[length(pops2$Stop)]  + 
                 0.5, y=0, xend= pops2$Stop[length(pops2$Stop)] + 0.5, yend=1,  alpha=0.9, size=0.25) +
  geom_segment(x=0.5, xend= pops2$Stop[length(pops2$Stop)]+0.5, y=1, yend=1, alpha=0.9, size=0.25) +
  geom_segment(x=0.5, xend= pops2$Stop[length(pops2$Stop)]+0.5, y=0, yend=0, alpha=0.9, size=0.25) +
  ylim(-0.1,1.01) +
  xlim(-0.1, pops2$Stop[length(pops2$Stop)]+1) +
  theme(panel.background = element_blank()) +
  xlab("") + theme(legend.position = "") +
 # theme(axis.text.x = element_blank(), axis.ticks.x=element_blank()) +
  ggtitle("K = 3") +
  theme(plot.title = element_text(hjust = 0.5, face = "bold")) +
  scale_x_continuous(breaks=pops2$Position, labels=pops2$Location) +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  scale_fill_viridis_d(option="B")


p3

ggsave("outputs/105/lfs-dsm-reference-k3.jpg")
```


```{r}
q4<-read_delim("outputs/105/260-K4.qopt", delim=" ", col_names=FALSE) %>% select(X1, X2, X3, X4) %>%
  rename(Q1=X1, Q2=X2, Q3=X3, Q4=X4)

df4<-bind_cols(q4, meta)            

q4s<-df4 %>% select(Location, Sample, Q1, Q2, Q3, Q4) %>% arrange(Location,Sample) %>%
  mutate(Index=1:n()) %>% gather(key=Ancestry, value=Q, 3:6) %>% mutate(K=4)

pops4<-q4s  %>% group_by(Location) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(Location,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2)) %>% ungroup() %>% unique()

```

```{r}
p4 <- ggplot(q4s) + 
  geom_col(aes(x=Index,y=Q, fill=Ancestry), color="NA", size = 0, width = 1) +
  geom_segment(data=pops2, x = pops2$Start - 0.5, y=0, xend = pops2$Start-0.5, yend=1, alpha=0.9, size=0.25) +
  geom_segment(data=pops2, x = pops2$Stop[length(pops2$Stop)]  + 
                 0.5, y=0, xend= pops2$Stop[length(pops2$Stop)] + 0.5, yend=1,  alpha=0.9, size=0.25) +
  geom_segment(x=0.5, xend= pops2$Stop[length(pops2$Stop)]+0.5, y=1, yend=1, alpha=0.9, size=0.25) +
  geom_segment(x=0.5, xend= pops2$Stop[length(pops2$Stop)]+0.5, y=0, yend=0, alpha=0.9, size=0.25) +
  ylim(-0.1,1.01) +
  xlim(-0.1, pops2$Stop[length(pops2$Stop)]+1) +
  theme(panel.background = element_blank()) +
  xlab("") + theme(legend.position = "") +
 # theme(axis.text.x = element_blank(), axis.ticks.x=element_blank()) +
  ggtitle("K = 4") +
  theme(plot.title = element_text(hjust = 0.5, face = "bold")) +
  scale_x_continuous(breaks=pops2$Position, labels=pops2$Location) +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  scale_fill_viridis_d(option="B")


p4

ggsave("outputs/105/lfs-dsm-reference-k4.jpg")
```

```{r}
q5<-read_delim("outputs/105/260-K5.qopt", delim=" ", col_names=FALSE) %>% select(X1, X2, X3, X4, X5) %>%
  rename(Q1=X1, Q2=X2, Q3=X3, Q4=X4, Q5=X5)

df5<-bind_cols(q5, meta)            

q5s<-df5 %>% select(Location, Sample, Q1, Q2, Q3, Q4, Q5) %>% arrange(Location,Sample) %>%
  mutate(Index=1:n()) %>% gather(key=Ancestry, value=Q, 3:7) %>% mutate(K=5)

pops5<-q5s  %>% group_by(Location) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(Location,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2)) %>% ungroup() %>% unique()

```

```{r}
p5 <- ggplot(q5s) + 
  geom_col(aes(x=Index,y=Q, fill=Ancestry), color="NA", size = 0, width = 1) +
  geom_segment(data=pops2, x = pops2$Start - 0.5, y=0, xend = pops2$Start-0.5, yend=1, alpha=0.9, size=0.25) +
  geom_segment(data=pops2, x = pops2$Stop[length(pops2$Stop)]  + 
                 0.5, y=0, xend= pops2$Stop[length(pops2$Stop)] + 0.5, yend=1,  alpha=0.9, size=0.25) +
  geom_segment(x=0.5, xend= pops2$Stop[length(pops2$Stop)]+0.5, y=1, yend=1, alpha=0.9, size=0.25) +
  geom_segment(x=0.5, xend= pops2$Stop[length(pops2$Stop)]+0.5, y=0, yend=0, alpha=0.9, size=0.25) +
  ylim(-0.1,1.01) +
  xlim(-0.1, pops2$Stop[length(pops2$Stop)]+1) +
  theme(panel.background = element_blank()) +
  xlab("") + theme(legend.position = "") +
 # theme(axis.text.x = element_blank(), axis.ticks.x=element_blank()) +
  ggtitle("K = 5") +
  theme(plot.title = element_text(hjust = 0.5, face = "bold")) +
  scale_x_continuous(breaks=pops2$Position, labels=pops2$Location) +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  scale_fill_viridis_d(option="B")


p5

ggsave("outputs/105/lfs-dsm-reference-k5.jpg")
```


#Isolate Nook
```{r}
nm<-meta %>% filter(Location !="New Sequencing")
write_tsv(nm %>%select(Path), file="bamlists/bamlist134.bamlist")
nm$Location
```

134 Samples

```{sh, eval=FALSE}
srun -p high -t 5:00:00 --mem=16G --nodes=1 $HOME/angsd/angsd -P 12 \
  -ref /home/maccamp/genomes/hypomesus-20210204/Hyp_tra_F_20210204.fa \
  -bam $HOME/longfin/bamlists/bamlist134.bamlist -GL 1 \
  -doGLF 2 -doMajorMinor 1 -doMaf 2 -SNP_pval 1e-6 -minMapQ 10 -minQ 20 -minMaf 0.05 \
   -rf $HOME/delta-smelt/metadata/large-contigs.txt  \
   -minInd 117 \
  -out $HOME/longfin/outputs/105/130-beagle > outputs/105/130-beagle-contigs.out 2> outputs/105/130-beagle-contigs.err &
  

python $HOME/pcangsd/pcangsd.py -beagle outputs/105/130-beagle.beagle.gz \
  -o outputs/105/130-pca -threads 10
```
