---
title: "105-integrating-data"
author: "Mac Campbell"
date: "2023-01-06"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
```

1. Really good to know Russian River LF sample #'s 22-0074 22-0102, only two fish total     

## Existing data

```{r}
existing<-read_tsv("meta/counts.tsv") %>% filter(Alignment !="Original")
existing<-existing %>% filter(Alignment=="New Genome") %>%
  mutate(Path=paste0("/home/maccamp/data/longfin/",File)) %>% filter(Reads > 1.5e5)

ggplot(existing) + geom_histogram(aes(x=Reads))
```

```{r}
existing %>% summarize(Mean=mean(Reads), Median=median(Reads), max(Reads))
  
```

## New Data

```{r}
dat<-read_csv("outputs/104/alt.dat", col_names = c("Sample","Sort","Dedup"))

ds<-dat %>% filter(Dedup > 1.5e5) %>% mutate(Path=paste0("data/alt-bams/",Sample,".sort.flt.bam"))

ggplot(ds) +
  geom_histogram(aes(x=Dedup))
```

```{r}
mm<-existing %>% select(-File) %>% rename(Dedup=Reads) %>% select(-Alignment) %>% mutate(Origin="Old")
mm %>% bind_rows(ds %>% select(-Sort) %>% mutate(Origin="New"))
```

## Combine by trimming then downsampling

```{r}
plate1<-dat[startsWith(dat$Sample, "HS2253"),] 
plate1<-plate1 %>% mutate(Plate="plate-1")
plate1$Barcode<-gsub("HS2253_AGTCAA-","",plate1$Sample)

plate23<-dat[(!startsWith(dat$Sample, "HS2253")),] 
plate23<-plate23 %>% separate(Sample, into = c("p1","p2","Barcode"), remove=FALSE) %>% mutate(Plate=paste0(p1,"-",p2)) %>% select(-p1,-p2)

plates<-bind_rows(plate1, plate23)

#HS2253_AGTCAA_RA_GGTGGCTTCATGCAGG.fastq

newseqs<- plates %>% select(-Sample) %>% mutate(Sample=paste0(Plate,"-",Barcode)) %>%
  mutate(PathRA = ifelse(Plate=="plate-1", paste0("data/alt-1/", "HS2253_AGTCAA_RA_GG", Barcode,"TGCAGG.fastq"),
                  ifelse(Plate=="plate-2", paste0("data/alt-2/", "HS2253_AGTCAA_RA_GG", Barcode,"TGCAGG.fastq"),
                  ifelse(Plate=="plate-3", paste0("data/alt-3/", "HS2253_AGTCAA_RA_GG", Barcode,"TGCAGG.fastq"),"NULL")))) %>%
  mutate(PathRB = ifelse(Plate=="plate-1", paste0("data/alt-1/", "HS2253_AGTCAA_RB_GG", Barcode,"TGCAGG.fastq"),
                  ifelse(Plate=="plate-2", paste0("data/alt-2/", "HS2253_AGTCAA_RB_GG", Barcode,"TGCAGG.fastq"),
                  ifelse(Plate=="plate-3", paste0("data/alt-3/", "HS2253_AGTCAA_RB_GG", Barcode,"TGCAGG.fastq"),"NULL")))) 
```


My forward are 135 bp in length theirs are 85, trimming 50 bp
My reverse are 151 bp in length, theirs are 101, trimming 50 bp

'module load seqtk'    

```{r}
rename<-newseqs %>% mutate(r1=paste0("seqtk trimfq -e 50 ", PathRA, " > data/alt-trim/", Sample,"_RA.fastq")) %>% 
                    mutate(r2=paste0("seqtk trimfq -e 50 ", PathRB, " > data/alt-trim/", Sample,"_RB.fastq"))
r1<- select(rename, r1) %>% rename(Command=r1)
r2<- select(rename, r2) %>% rename(Command=r2)
r3 <- bind_rows(r1, r2)
r3
write_tsv(r3, "105.1-batch-trim.sh",col_names = FALSE)
```

srun -p bigmemh -t 02:00:00 --nodes=1 parallel -j 10 < 105.1-batch-trim.sh > outputs/105/trim.stdout 2> outputs/105/trim.stderr 

Trim and re-align
```{sh, eval=FALSE}
ls | grep RA | perl -pe 's/.fastq//g' > forward
ls | grep RB | perl -pe 's/.fastq//g' > reverse
ls | grep RA | perl -pe 's/_RA.fastq//g' > name
paste forward reverse name  > samples.txt
bash $HOME/missouri-trout/doAlign-unzipped.sh samples.txt /home/maccamp/genomes/hypomesus-20210204/Hyp_tra_F_20210204.fa 
```

Then downsample!!     


```{r}
reads<-read_csv("outputs/105/alt-trim-reads.txt", col_names=c("Sample","Aligned","Dedup"))
m2<-rename %>% select(-Sort, -Dedup, -PathRA, -PathRB, -r1, -r2) %>% left_join(reads)
m2 %>% filter(Dedup > 1e5) %>% 
  summarize(Mean=mean(Aligned), Median=median(Dedup))
```

```{r}
ggplot(m2 %>% filter(Dedup > 1e5)) +
   geom_histogram(aes(x=Dedup))
```

Downsample to 1e6

```{r}
m3<-m2 %>% filter(Dedup > 1e5) %>% mutate(Frac=1e6/Dedup)  %>% 
  mutate(Path=ifelse(Dedup > 1e6, paste0("data/alt-trim/",Sample,".reduced.bam" ),
                paste0("data/alt-trim/",Sample,".sort.flt.bam")))

downsample<-m3 %>% filter(Dedup > 1e6 ) %>%
  mutate(ReductionCommand = paste0("samtools view -bs ",Frac, " ", "/home/maccamp/longfin/data/alt-trim/",
                                   Sample, ".sort.flt.bam"," > ",
                                   "/home/maccamp/longfin/data/alt-trim/",
                                   Sample,".reduced.bam" )) 

write_csv(downsample$ReductionCommand %>% as_tibble(), "105.1-downsample.sh", col_names = FALSE)


m3
```

srun -p high -t 02:00:00 --nodes=1 parallel -j 10 < 105.1-downsample.sh > outputs/105/downsample.stdout 2> outputs/105/downsample.stderr 

```{r}
m192<-m3 %>% filter(Dedup > 2e5)
write_tsv(m192 %>% select(Path), col_names = FALSE, "bamlists/192.bamlist")

names<-read_tsv("meta/non-lfs.tsv")
names$Sample<-gsub("HS2253_AGTCAA","plate-1",names$Sample)
m155<-m3 %>% filter(!(Sample %in% names$Sample)) 

write_tsv(m155 %>% select(Path), col_names = FALSE, "bamlists/155.bamlist")

```

     
     
```{sh, eval=FALSE}
srun -p high -t 14:00:00 --mem=16G --nodes=1 $HOME/angsd/angsd -P 12 \
  -ref /home/maccamp/genomes/hypomesus-20210204/Hyp_tra_F_20210204.fa \
  -bam $HOME/longfin/bamlists/155.bamlist -GL 1 \
  -doGLF 2 -doMajorMinor 1 -doMaf 2 -SNP_pval 1e-6 -minMapQ 10 -minQ 20 -minMaf 0.05 \
  -r lg01 \
   -minInd 116 \
  -out $HOME/longfin/outputs/105/155-beagle > outputs/105/beagle-contigs.out 2> outputs/105/beagle-contigs.err &
```

-rf $HOME/delta-smelt/metadata/large-contigs.txt  
   
-r lg01 9420 sites with all species
-r lg01 2241 sites with LFS

```{sh, eval=FALSE}
python $HOME/pcangsd/pcangsd.py -beagle outputs/105/155-beagle.beagle.gz \
   -admix -o outputs/105/155-pca -threads 10
```




```{r}
cov<-read_delim("outputs/105/192-pca.cov", col_names=FALSE, delim=" ") %>% as.matrix()
```

```{r}
#' @param samples character vector with the individuals IDs in the order in which
#' they were passed in the bamlist to angsd.
#' @param cov covariance matrix
covar2pcs <- function(samples, cov) {
  
  
  eig <- eigen(cov, symm = TRUE)
  PC <- as.data.frame(eig$vectors) %>%
    as_tibble() %>%
    setNames(sprintf("PC-%02d", 1:ncol(.)))
  
  samtib <- tibble(sample = samples)
  
  list(
    PCs = bind_cols(samtib, PC),
    eigevalues = eig$values
  )
}
```

```{r}
meta<-m192
pca <- covar2pcs(meta$Sample, cov)

pca_long <- pca$PCs %>%
  tidyr::gather(., key = "PC", "val", -sample)

# then expand a grid of the possible comparisons (ordered)
expg <- expand.grid(sample = pca$PCs$sample,
                    PCx = sprintf("PC-%02d", 1:6),
                    PCy = sprintf("PC-%02d", 1:6),
                    stringsAsFactors = FALSE) %>%
  tibble::as_tibble()

# then left join the pca results onto that
pca_pairs <- dplyr::left_join(expg, pca_long, by = c("sample", "PCx" = "PC")) %>%
  dplyr::rename(val_x = val) %>%
  dplyr::left_join(pca_long, by = c("sample", "PCy" = "PC")) %>%
  dplyr::rename(val_y = val)

pp_meta <- pca_pairs %>%   # just keep the first 6 PCs around
  left_join(., meta, by = c("sample" = "Sample")) 
```

Plot    

```{r}
npc <- 3
pp_meta2 <- pp_meta %>%
  filter( (PCx %in% sprintf("PC-%02d", 1:npc)) & 
            (PCy %in% sprintf("PC-%02d", 1:npc)) )

eig <- eigen(cov, symm = TRUE)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```

```{r}
ggplot(pp_meta2, aes(x = val_x, y = val_y)) +
  geom_point() +
  facet_grid(PCx ~ PCy)
```

## Merging

```{r}
e2<-existing %>% filter(Reads > 2e5)%>% select(-File) %>% 
  rename(Dedup=Reads) %>% select(-Alignment) %>% 
  separate(Sample, into = c("Location","Number"), remove=FALSE) %>% select(-Number)

m140<-m155 %>% filter(Dedup > 2e5) %>% select(Sample, Dedup, Path) %>% mutate(Location="New Sequencing")

c<-bind_rows(e2,m140)
write_tsv(select(c,Path), "bamlists/283.bamlist",col_names = FALSE)
```

```{sh, eval=FALSE}
srun -p high -t 14:00:00 --mem=16G --nodes=1 $HOME/angsd/angsd -P 12 \
  -ref /home/maccamp/genomes/hypomesus-20210204/Hyp_tra_F_20210204.fa \
  -bam $HOME/longfin/bamlists/2813.bamlist -GL 1 \
  -doGLF 2 -doMajorMinor 1 -doMaf 2 -SNP_pval 1e-6 -minMapQ 10 -minQ 20 -minMaf 0.05 \
  -r lg01 \
   -minInd 212 \
  -out $HOME/longfin/outputs/105/283-beagle > outputs/105/beagle-contigs.out 2> outputs/105/beagle-contigs.err &
  ```