---
title: "105-integrating-data"
author: "Mac Campbell"
date: "2023-01-06"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
```

1. Really good to know Russian River LF sample #'s 22-0074 22-0102, only two fish total     

## Existing data

```{r}
existing<-read_tsv("meta/counts.tsv") %>% filter(Alignment !="Original")
existing<-existing %>% filter(Alignment=="New Genome") %>%
  mutate(Path=paste0("/home/maccamp/data/longfin/",File)) %>% filter(Reads > 1.5e5)

ggplot(existing) + geom_histogram(aes(x=Reads))
```

## New Data

```{r}
dat<-read_csv("outputs/104/alt.dat", col_names = c("Sample","Sort","Dedup"))

ds<-dat %>% filter(Dedup > 1.5e5) %>% mutate(Path=paste0("data/alt-bams/",Sample,".sort.flt.bam"))

ggplot(ds) +
  geom_histogram(aes(x=Dedup))
```

```{r}
mm<-existing %>% select(-File) %>% rename(Dedup=Reads) %>% select(-Alignment) %>% mutate(Origin="Old")
mm %>% bind_rows(ds %>% select(-Sort) %>% mutate(Origin="New"))
```

## Combine by trimming then downsampling

```{r}
plate1<-dat[startsWith(dat$Sample, "HS2253"),] 
plate1<-plate1 %>% mutate(Plate="plate-1")
plate1$Barcode<-gsub("HS2253_AGTCAA-","",plate1$Sample)

plate23<-dat[(!startsWith(dat$Sample, "HS2253")),] 
plate23<-plate23 %>% separate(Sample, into = c("p1","p2","Barcode"), remove=FALSE) %>% mutate(Plate=paste0(p1,"-",p2)) %>% select(-p1,-p2)

plates<-bind_rows(plate1, plate23)

#HS2253_AGTCAA_RA_GGTGGCTTCATGCAGG.fastq

newseqs<- plates %>% select(-Sample) %>% mutate(Sample=paste0(Plate,"-",Barcode)) %>%
  mutate(PathRA = ifelse(Plate=="plate-1", paste0("data/alt-1/", "HS2253_AGTCAA_RA_GG", Barcode,"TGCAGG.fastq"),
                  ifelse(Plate=="plate-2", paste0("data/alt-2/", "HS2253_AGTCAA_RA_GG", Barcode,"TGCAGG.fastq"),
                  ifelse(Plate=="plate-3", paste0("data/alt-3/", "HS2253_AGTCAA_RA_GG", Barcode,"TGCAGG.fastq"),"NULL")))) %>%
  mutate(PathRB = ifelse(Plate=="plate-1", paste0("data/alt-1/", "HS2253_AGTCAA_RB_GG", Barcode,"TGCAGG.fastq"),
                  ifelse(Plate=="plate-2", paste0("data/alt-2/", "HS2253_AGTCAA_RB_GG", Barcode,"TGCAGG.fastq"),
                  ifelse(Plate=="plate-3", paste0("data/alt-3/", "HS2253_AGTCAA_RB_GG", Barcode,"TGCAGG.fastq"),"NULL")))) 
```


My forward are 135 bp in length theirs are 85, trimming 50 bp
My reverse are 151 bp in length, theirs are 101, trimming 50 bp

'module load seqtk'    

```{r}
rename<-newseqs %>% mutate(r1=paste0("seqtk trimfq -e 50 ", PathRA, " > data/alt-trim/", Sample,"_RA.fastq")) %>% 
                    mutate(r2=paste0("seqtk trimfq -e 50 ", PathRB, " > data/alt-trim/", Sample,"_RB.fastq"))
r1<- select(rename, r1) %>% rename(Command=r1)
r2<- select(rename, r2) %>% rename(Command=r2)
r3 <- bind_rows(r1, r2)
r3
write_tsv(r3, "105.1-batch-trim.sh",col_names = FALSE)
```

srun -p bigmemh --nodes=1 parallel -j 10 < 105.1-batch-trim.sh > outputs/105/trim.stdout 2> outputs/105/trim.stderr 

Trim and re-align
```{sh, eval=FALSE}
ls | grep RA | perl -pe 's/.fastq//g' > forward
ls | grep RB | perl -pe 's/.fastq//g' > reverse
ls | grep RA | perl -pe 's/_RA.fastq//g' > name
paste forward reverse name  > samples.txt
bash $HOME/missouri-trout/doAlign-unzipped.sh samples.txt /home/maccamp/genomes/hypomesus-20210204/Hyp_tra_F_20210204.fa 
```
