---
title: "106-new-seq"
output: html_document
date: "2023-04-28"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
library(ggrepel)
library(ggpubr)
```

```{r}
meta<-read_csv("meta/meta260.csv")
meta$Location<-factor(meta$Location, levels=c("YBAK","SKNA","PTLC","HRLC","FRAS","LWSH",
                                              "NOOK","COLR",
                                              "HUMB","PETA","ALVS","SUIB","CHPI",
                                         "New Sequencing"))

meta

write_tsv(meta %>% select(Sample), file="bamlists/260.names",col_names = FALSE)


m154<-meta %>% group_by(Location) %>% top_n(20, Dedup)
write_tsv(m154 %>% select(Path), file="bamlists/154.bamlist",col_names = FALSE)
write_tsv(m154 %>% select(Sample), file="bamlists/154.names",col_names = FALSE)

```
## Have seqs

Will try to control for effects a bit. Making a region file from 101 to restrict snps to previously discovered ones. Using trimmed and downsampled sequences from 105.

(base) maccamp@farm:~/longfin/outputs/101$ gunzip -c sub140-contigs.beagle.gz | cut -f 1 | perl -pe 's/_/:/g' > ../106/regions.txt
grep lg01 regions.txt > lg01.txt
grep ^lg regions.txt > lgs.txt
60639 sites

```{eval=FALSE}
srun -p high -t 14:00:00 --mem=16G --nodes=1 $HOME/angsd/angsd -P 8 \
  -ref /home/maccamp/genomes/hypomesus-20210204/Hyp_tra_F_20210204.fa \
  -bam $HOME/longfin/bamlists/260.bamlist -GL 1 \
  -doGLF 2 -doMajorMinor 1 -doMaf 2 -SNP_pval 1e-6 -minMapQ 10 -minQ 20 -minMaf 0.05 \
  -rf $HOME/longfin/outputs/106/lg01.txt -minInd 208 \
  -out $HOME/longfin/outputs/106/260-txt > outputs/106/beagle.out 2> outputs/106/beagle.err &

  #1674 snps
  
srun -p high -t 14:00:00 --mem=16G --nodes=1 $HOME/angsd/angsd -P 8 \
  -ref /home/maccamp/genomes/hypomesus-20210204/Hyp_tra_F_20210204.fa \
  -bam $HOME/longfin/bamlists/260.bamlist -GL 1 \
  -doGLF 2 -doMajorMinor 1 -doMaf 2 -SNP_pval 1e-6 -minMapQ 10 -minQ 20 \
  -rf $HOME/longfin/outputs/106/lg01.txt -minInd 208 \
  -out $HOME/longfin/outputs/106/260-no-maf > outputs/106/beagle-maf.out 2> outputs/106/beagle-maf.err &  
  
python $HOME/pcangsd/pcangsd.py -beagle outputs/106/260-txt.beagle.gz -o outputs/106/260-lg01 -threads 10
#1674 sites without MAF filtercs 

srun -p high -t 32:00:00 --mem=16G --nodes=1 $HOME/angsd/angsd -P 8 \
  -ref /home/maccamp/genomes/hypomesus-20210204/Hyp_tra_F_20210204.fa \
  -bam $HOME/longfin/bamlists/260.bamlist -GL 1 \
  -doGLF 2 -doMajorMinor 1 -doMaf 2 -SNP_pval 1e-6 -minMapQ 10 -minQ 20 -minMaf 0.05 \
  -rf $HOME/longfin/outputs/106/lgs.txt -minInd 208 \
  -out $HOME/longfin/outputs/106/260-lgs > outputs/106/beagle-lgs.out 2> outputs/106/beagle-lgs.err &

#51259 sites
  
conda activate py2
srun -p high -t 01:00:00 python $HOME/pcangsd/pcangsd.py -beagle outputs/106/260-lgs.beagle.gz \
  -o outputs/106/260-lgs -threads 10 
conda deactivate

```


```{r}
#' @param samples character vector with the individuals IDs in the order in which
#' they were passed in the bamlist to angsd.
#' @param cov covariance matrix
covar2pcs <- function(samples, cov) {
  
  
  eig <- eigen(cov, symm = TRUE)
  PC <- as.data.frame(eig$vectors) %>%
    as_tibble() %>%
    setNames(sprintf("PC-%02d", 1:ncol(.)))
  
  samtib <- tibble(sample = samples)
  
  list(
    PCs = bind_cols(samtib, PC),
    eigevalues = eig$values
  )
}
```

```{r}
#cov<-read_delim("outputs/106/260-lg01.cov", col_names=FALSE, delim=" ") %>% as.matrix()
cov<-read_delim("outputs/106/260-lgs.cov", col_names=FALSE, delim=" ") %>% as.matrix()
```

```{r}
pca <- covar2pcs(meta$Sample, cov)

pca_long <- pca$PCs %>%
  tidyr::gather(., key = "PC", "val", -sample)

# then expand a grid of the possible comparisons (ordered)
expg <- expand.grid(sample = pca$PCs$sample,
                    PCx = sprintf("PC-%02d", 1:6),
                    PCy = sprintf("PC-%02d", 1:6),
                    stringsAsFactors = FALSE) %>%
  tibble::as_tibble()

# then left join the pca results onto that
pca_pairs <- dplyr::left_join(expg, pca_long, by = c("sample", "PCx" = "PC")) %>%
  dplyr::rename(val_x = val) %>%
  dplyr::left_join(pca_long, by = c("sample", "PCy" = "PC")) %>%
  dplyr::rename(val_y = val)

pp_meta <- pca_pairs %>%   # just keep the first 6 PCs around
  left_join(., meta, by = c("sample" = "Sample")) 
```


```{r}
npc <- 3
pp_meta2 <- pp_meta %>%
  filter( (PCx %in% sprintf("PC-%02d", 1:npc)) & 
            (PCy %in% sprintf("PC-%02d", 1:npc)) ) %>%
  mutate(Origin=ifelse(Location=="New Sequencing","New","Old"))

eig <- eigen(cov, symm = TRUE)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```

```{r}
ggplot(pp_meta2, aes(x = val_x, y = val_y)) +
  geom_point(aes(color=Location)) + 
  facet_grid(PCx ~ PCy)
```

```{r}
sub12<-pp_meta2 %>% filter( (PCx =="PC-01") & (PCy =="PC-02") )
text<- sub12 %>% select(sample,Location,val_x,val_y) %>% group_by(Location) %>%
  mutate(meanX=mean(val_x), meanY=mean(val_y)) %>% select(Location, meanX, meanY) %>%
  unique()   

pc12<-ggplot(sub12, aes(x = val_x, y = val_y, color=Location))+
  geom_point(size = 2, alpha=0.75) +
  geom_label_repel(data=text, aes(x=meanX, y=meanY, color=Location,
                                  label=Location), max.overlaps = Inf) +
  scale_fill_discrete(na.value = "white") + 
  theme_bw()+
  theme(panel.grid=element_blank())+
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = ""))+
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = ""))+
  theme(legend.position = "")+
  theme(plot.title = element_text(hjust = 0.5)) 

sub13<-pp_meta2 %>% filter( (PCx =="PC-01") & (PCy =="PC-03") )
text13<- sub13 %>% select(sample,Location,val_x,val_y) %>% group_by(Location) %>%
  mutate(meanX=mean(val_x), meanY=mean(val_y)) %>% select(Location, meanX, meanY) %>%
  unique()   

pc13<-ggplot(sub13, aes(x = val_x, y = val_y, color = Location)) +
  geom_point(size = 2, alpha=0.75) +
  geom_label_repel(data=text13, aes(x=meanX, y=meanY, color=Location,
                                  label=Location), max.overlaps = Inf) +
  scale_fill_discrete(na.value = "white") + 
  theme_bw()+
  theme(panel.grid=element_blank())+
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = ""))+
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = ""))+
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(legend.position = "none")

pcs<-ggarrange(pc12, pc13, ncol= 2, widths=c(1,1))
pcs
ggsave("outputs/106/combined-pca-260.jpeg", width=10)
```

```{sh, eval=FALSE}
srun -p high -t 01:00:00 --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/106/260-lgs.beagle.gz -K 2 -o outputs/106/260-lgs.K2 &
srun -p high -t 01:00:00 --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/106/260-lgs.beagle.gz -K 3 -o outputs/106/260-lgs.K3 &
srun -p  high -t 01:00:00 --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/106/260-lgs.beagle.gz -K 4 -o outputs/106/260-lgs.K4 &
srun -p high -t 01:00:00 --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/106/260-lgs.beagle.gz -K 5 -o outputs/106/260-lgs.K5 &

```



```{r}
q2<-read_delim("outputs/106/260-lgs.K2.qopt", delim=" ", col_names=FALSE) %>% select(X1, X2) %>%
  rename(Q1=X1, Q2=X2)

df2<-bind_cols(q2, meta)            

q2s<-df2 %>% select(Location, Sample, Q1, Q2) %>% arrange(Location,Sample) %>%
  mutate(Index=1:n()) %>% gather(key=Ancestry, value=Q, 3:4) %>% mutate(K=2)

pops2<-q2s  %>% group_by(Location) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(Location,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2)) %>% ungroup() %>% unique()

```

```{r}
p2 <- ggplot(q2s) + 
  geom_col(aes(x=Index,y=Q, fill=Ancestry), color="NA", size = 0, width = 1) +
  geom_segment(data=pops2, x = pops2$Start - 0.5, y=0, xend = pops2$Start-0.5, yend=1, alpha=0.9, size=0.25) +
  geom_segment(data=pops2, x = pops2$Stop[length(pops2$Stop)]  + 
                 0.5, y=0, xend= pops2$Stop[length(pops2$Stop)] + 0.5, yend=1,  alpha=0.9, size=0.25) +
  geom_segment(x=0.5, xend= pops2$Stop[length(pops2$Stop)]+0.5, y=1, yend=1, alpha=0.9, size=0.25) +
  geom_segment(x=0.5, xend= pops2$Stop[length(pops2$Stop)]+0.5, y=0, yend=0, alpha=0.9, size=0.25) +
  ylim(-0.1,1.01) +
  xlim(-0.1, pops2$Stop[length(pops2$Stop)]+1) +
  theme(panel.background = element_blank()) +
  xlab("") + theme(legend.position = "") +
 # theme(axis.text.x = element_blank(), axis.ticks.x=element_blank()) +
  ggtitle("K = 2") +
  theme(plot.title = element_text(hjust = 0.5, face = "bold")) +
  scale_x_continuous(breaks=pops2$Position, labels=pops2$Location) +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  scale_fill_viridis_d(option="B")


p2

ggsave("outputs/106/lfs-dsm-reference-k2.jpg")
```

```{r}
q3<-read_delim("outputs/106/260-lgs.K3.qopt", delim=" ", col_names=FALSE) %>% select(X1, X2, X3) %>%
  rename(Q1=X1, Q2=X2, Q3=X3)

df3<-bind_cols(q3, meta)            

q3s<-df3 %>% select(Location, Sample, Q1, Q2, Q3) %>% arrange(Location,Sample) %>%
  mutate(Index=1:n()) %>% gather(key=Ancestry, value=Q, 3:5) %>% mutate(K=3)

pops3<-q3s  %>% group_by(Location) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(Location,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2)) %>% ungroup() %>% unique()

```

```{r}
p3 <- ggplot(q3s) + 
  geom_col(aes(x=Index,y=Q, fill=Ancestry), color="NA", size = 0, width = 1) +
  geom_segment(data=pops2, x = pops2$Start - 0.5, y=0, xend = pops2$Start-0.5, yend=1, alpha=0.9, size=0.25) +
  geom_segment(data=pops2, x = pops2$Stop[length(pops2$Stop)]  + 
                 0.5, y=0, xend= pops2$Stop[length(pops2$Stop)] + 0.5, yend=1,  alpha=0.9, size=0.25) +
  geom_segment(x=0.5, xend= pops2$Stop[length(pops2$Stop)]+0.5, y=1, yend=1, alpha=0.9, size=0.25) +
  geom_segment(x=0.5, xend= pops2$Stop[length(pops2$Stop)]+0.5, y=0, yend=0, alpha=0.9, size=0.25) +
  ylim(-0.1,1.01) +
  xlim(-0.1, pops2$Stop[length(pops2$Stop)]+1) +
  theme(panel.background = element_blank()) +
  xlab("") + theme(legend.position = "") +
 # theme(axis.text.x = element_blank(), axis.ticks.x=element_blank()) +
  ggtitle("K = 3") +
  theme(plot.title = element_text(hjust = 0.5, face = "bold")) +
  scale_x_continuous(breaks=pops2$Position, labels=pops2$Location) +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  scale_fill_viridis_d(option="B")


p3

ggsave("outputs/106/lfs-dsm-reference-k3.jpg")
```


```{r}
q4<-read_delim("outputs/106/260-lgs.K4.qopt", delim=" ", col_names=FALSE) %>% select(X1, X2, X3, X4) %>%
  rename(Q1=X1, Q2=X2, Q3=X3, Q4=X4)

df4<-bind_cols(q4, meta)            

q4s<-df4 %>% select(Location, Sample, Q1, Q2, Q3, Q4) %>% arrange(Location,Sample) %>%
  mutate(Index=1:n()) %>% gather(key=Ancestry, value=Q, 3:6) %>% mutate(K=4)

pops4<-q4s  %>% group_by(Location) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(Location,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2)) %>% ungroup() %>% unique()

```

```{r}
p4 <- ggplot(q4s) + 
  geom_col(aes(x=Index,y=Q, fill=Ancestry), color="NA", size = 0, width = 1) +
  geom_segment(data=pops2, x = pops2$Start - 0.5, y=0, xend = pops2$Start-0.5, yend=1, alpha=0.9, size=0.25) +
  geom_segment(data=pops2, x = pops2$Stop[length(pops2$Stop)]  + 
                 0.5, y=0, xend= pops2$Stop[length(pops2$Stop)] + 0.5, yend=1,  alpha=0.9, size=0.25) +
  geom_segment(x=0.5, xend= pops2$Stop[length(pops2$Stop)]+0.5, y=1, yend=1, alpha=0.9, size=0.25) +
  geom_segment(x=0.5, xend= pops2$Stop[length(pops2$Stop)]+0.5, y=0, yend=0, alpha=0.9, size=0.25) +
  ylim(-0.1,1.01) +
  xlim(-0.1, pops2$Stop[length(pops2$Stop)]+1) +
  theme(panel.background = element_blank()) +
  xlab("") + theme(legend.position = "") +
 # theme(axis.text.x = element_blank(), axis.ticks.x=element_blank()) +
  ggtitle("K = 4") +
  theme(plot.title = element_text(hjust = 0.5, face = "bold")) +
  scale_x_continuous(breaks=pops2$Position, labels=pops2$Location) +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  scale_fill_viridis_d(option="B")


p4

ggsave("outputs/106/lfs-dsm-reference-k4.jpg")
```

```{r}
q5<-read_delim("outputs/106/260-lgs.K5.qopt", delim=" ", col_names=FALSE) %>% select(X1, X2, X3, X4, X5) %>%
  rename(Q1=X1, Q2=X2, Q3=X3, Q4=X4, Q5=X5)

df5<-bind_cols(q5, meta)            

q5s<-df5 %>% select(Location, Sample, Q1, Q2, Q3, Q4, Q5) %>% arrange(Location,Sample) %>%
  mutate(Index=1:n()) %>% gather(key=Ancestry, value=Q, 3:7) %>% mutate(K=5)

pops5<-q5s  %>% group_by(Location) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(Location,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2)) %>% ungroup() %>% unique()

```

```{r}
p5 <- ggplot(q5s) + 
  geom_col(aes(x=Index,y=Q, fill=Ancestry), color="NA", size = 0, width = 1) +
  geom_segment(data=pops2, x = pops2$Start - 0.5, y=0, xend = pops2$Start-0.5, yend=1, alpha=0.9, size=0.25) +
  geom_segment(data=pops2, x = pops2$Stop[length(pops2$Stop)]  + 
                 0.5, y=0, xend= pops2$Stop[length(pops2$Stop)] + 0.5, yend=1,  alpha=0.9, size=0.25) +
  geom_segment(x=0.5, xend= pops2$Stop[length(pops2$Stop)]+0.5, y=1, yend=1, alpha=0.9, size=0.25) +
  geom_segment(x=0.5, xend= pops2$Stop[length(pops2$Stop)]+0.5, y=0, yend=0, alpha=0.9, size=0.25) +
  ylim(-0.1,1.01) +
  xlim(-0.1, pops2$Stop[length(pops2$Stop)]+1) +
  theme(panel.background = element_blank()) +
  xlab("") + theme(legend.position = "") +
 # theme(axis.text.x = element_blank(), axis.ticks.x=element_blank()) +
  ggtitle("K = 5") +
  theme(plot.title = element_text(hjust = 0.5, face = "bold")) +
  scale_x_continuous(breaks=pops2$Position, labels=pops2$Location) +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  scale_fill_viridis_d(option="B")


p5

ggsave("outputs/106/lfs-dsm-reference-k5.jpg")
```

## Phylogeny

Rooting....? Let's try rooting with night smelt 

```{r}

night<-read_csv("meta/non-lfs-meta.csv") %>% filter(Species %in% c("Whitebait Smelt","Night Smelt")) %>% rename(Loc=Species) %>% mutate(Location=Loc) %>% select(Sample, Location, Dedup, Path, Loc) %>% filter(Dedup > 6e5)

phylo<-bind_rows(meta,night)
write_tsv(phylo %>% select(Path), col_names =FALSE, file="bamlists/phylo271.bamlist")
write_tsv(phylo %>% select(Sample), col_names =FALSE, file="bamlists/phylo271.names")

```

Gives us 11 fish. 271 total
-rf $HOME/longfin/outputs/106/lgs.txt  \

```{sh, eval=FALSE}
srun -p high -t 02:00:00 --mem=16G --nodes=1 --ntasks=1 angsd -P 12 \
-bam bamlists/phylo271.bamlist \
-out /home/maccamp/longfin/outputs/106/plink-lg01 \
-anc /home/maccamp/genomes/hypomesus-20210204/Hyp_tra_F_20210204.fa \
-r lg01 \
-minInd 244 -minMaf 0.05  -minMapQ 10 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 4 -doPost 1 -postCutoff 0.95 -doPlink 2 >outputs/106/plink.out 2>outputs/106/plink.stderr &

#342 sites

srun -p high -t 12:00:00 --mem=16G --nodes=1 --ntasks=1 angsd -P 12 \
-bam bamlists/phylo271.bamlist \
-out /home/maccamp/longfin/outputs/106/plink \
-anc /home/maccamp/genomes/hypomesus-20210204/Hyp_tra_F_20210204.fa \
-rf $HOME/delta-smelt/metadata/large-contigs.txt  \
-minInd 244 -minMaf 0.05  -minMapQ 10 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 4 -doPost 1 -postCutoff 0.95 -doPlink 2 >outputs/106/plink.out 2>outputs/106/plink.stderr &


#no ogs, 260 individuals

srun -p high -t 12:00:00 --mem=16G --nodes=1 --ntasks=1 angsd -P 12 \
-bam bamlists/260.bamlist \
-out /home/maccamp/longfin/outputs/106/plink-260 \
-anc /home/maccamp/genomes/hypomesus-20210204/Hyp_tra_F_20210204.fa \
-rf $HOME/delta-smelt/metadata/large-contigs.txt  \
-minInd 234 -minMaf 0.05  -minMapQ 10 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 4 -doPost 1 -postCutoff 0.95 -doPlink 2 >outputs/106/plink-260.out 2>outputs/106/plink-260.stderr &

#next, can restrict to certain sites

srun -p high -t 38:00:00 --mem=16G --nodes=1 --ntasks=1 angsd -P 12 \
-bam bamlists/phylo271.bamlist \
-out /home/maccamp/longfin/outputs/106/plink-sites \
-anc /home/maccamp/genomes/hypomesus-20210204/Hyp_tra_F_20210204.fa \
-rf $HOME/longfin/outputs/106/lgs.txt  \
-minInd 244 -minMaf 0.05  -minMapQ 10 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 4 -doPost 1 -postCutoff 0.95 -doPlink 2 >outputs/106/plink.out 2>outputs/106/plink.stderr &

```

```{sh, eval=FALSE}
plink --tped plink-260.tped --tfam plink-260.tfam  --out binary --recode --allow-extra-chr --noweb
plink --ped binary.ped --map binary.map --recode vcf --allow-extra-chr -out recode
module load bcftools
bcftools +prune -m 0.6 -w 10000 recode.vcf  -Ov -o recode.prune.vcf
bcftools reheader --samples ../../bamlists/260.names -o recode.prune.reheadered.vcf recode.prune.vcf

#bcftools reheader --samples ../../bamlists/phylo271.names -o recode.prune.reheadered.vcf recode.prune.vcf

source activate py2; ~/mccloud-rrt/vcf2phylip.py -i recode.prune.reheadered.vcf; conda deactivate;

#mv recode.prune.reheadered.min4.phy lg01.phy
```


dat<-read.nexus.data(file="outputs/202/svdq/259-svdq.nex")
write.nexus.data(dat, file="outputs/203/259-forsplits.nex")
## With NOOK only

```{sh, eval=FALSE}
srun -p med -t 1-12:00:00 --mem=16G --nodes=1 $HOME/angsd/angsd -P 22 \
  -ref /home/maccamp/genomes/hypomesus-20210204/Hyp_tra_F_20210204.fa \
  -bam $HOME/longfin/bamlists/bamlist134.bamlist -GL 1 \
  -doGLF 2 -doMajorMinor 1 -doMaf 2 -SNP_pval 1e-6 -minMapQ 10 -minQ 20 -minMaf 0.05 \
  -rf $HOME/longfin/outputs/106/lgs.txt -minInd 117 \
  -out $HOME/longfin/outputs/106/134-lgs > outputs/106/134-beagle-lgs.out 2> outputs/106/134-beagle-lgs.err &

#conda clean --all 
#conda create -n py2 python=2.7 scipy numpy pandas

#conda activate python2
# actual used pip to install scipy

#Both worked..
#42735 sites
python $HOME/pcangsd/pcangsd.py -beagle outputs/106/134-lgs.beagle.gz -o outputs/106/134-lgs -threads 10

```



```{r}
cov<-read_delim("outputs/106/134-lgs.cov", col_names=FALSE, delim=" ") %>% as.matrix()
m134<-meta %>% filter(Location != "New Sequencing")
```

```{r}
pca <- covar2pcs(m134$Sample, cov)

pca_long <- pca$PCs %>%
  tidyr::gather(., key = "PC", "val", -sample)

# then expand a grid of the possible comparisons (ordered)
expg <- expand.grid(sample = pca$PCs$sample,
                    PCx = sprintf("PC-%02d", 1:6),
                    PCy = sprintf("PC-%02d", 1:6),
                    stringsAsFactors = FALSE) %>%
  tibble::as_tibble()

# then left join the pca results onto that
pca_pairs <- dplyr::left_join(expg, pca_long, by = c("sample", "PCx" = "PC")) %>%
  dplyr::rename(val_x = val) %>%
  dplyr::left_join(pca_long, by = c("sample", "PCy" = "PC")) %>%
  dplyr::rename(val_y = val)

pp_meta <- pca_pairs %>%   # just keep the first 6 PCs around
  left_join(., m134, by = c("sample" = "Sample")) 
```


```{r}
npc <- 3
pp_meta2 <- pp_meta %>%
  filter( (PCx %in% sprintf("PC-%02d", 1:npc)) & 
            (PCy %in% sprintf("PC-%02d", 1:npc)) ) %>%
  mutate(Origin=ifelse(Location=="New Sequencing","New","Old"))

eig <- eigen(cov, symm = TRUE)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```

```{r}
ggplot(pp_meta2, aes(x = val_x, y = val_y)) +
  geom_point(aes(color=Location)) + 
  facet_grid(PCx ~ PCy)
```

```{r}
sub12<-pp_meta2 %>% filter( (PCx =="PC-01") & (PCy =="PC-02") )
text<- sub12 %>% select(sample,Location,val_x,val_y) %>% group_by(Location) %>%
  mutate(meanX=mean(val_x), meanY=mean(val_y)) %>% select(Location, meanX, meanY) %>%
  unique()   

pc12<-ggplot(sub12, aes(x = val_x, y = val_y, color=Location))+
  geom_point(size = 2, alpha=0.75) +
  geom_label_repel(data=text, aes(x=meanX, y=meanY, color=Location,
                                  label=Location), max.overlaps = Inf) +
  scale_fill_discrete(na.value = "white") + 
  theme_bw()+
  theme(panel.grid=element_blank())+
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = ""))+
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = ""))+
  theme(legend.position = "")+
  theme(plot.title = element_text(hjust = 0.5)) 

sub13<-pp_meta2 %>% filter( (PCx =="PC-01") & (PCy =="PC-03") )
text13<- sub13 %>% select(sample,Location,val_x,val_y) %>% group_by(Location) %>%
  mutate(meanX=mean(val_x), meanY=mean(val_y)) %>% select(Location, meanX, meanY) %>%
  unique()   

pc13<-ggplot(sub13, aes(x = val_x, y = val_y, color = Location)) +
  geom_point(size = 2, alpha=0.75) +
  geom_label_repel(data=text13, aes(x=meanX, y=meanY, color=Location,
                                  label=Location), max.overlaps = Inf) +
  scale_fill_discrete(na.value = "white") + 
  theme_bw()+
  theme(panel.grid=element_blank())+
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = ""))+
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = ""))+
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(legend.position = "none")

pcs<-ggarrange(pc12, pc13, ncol= 2, widths=c(1,1))
pcs
ggsave("outputs/106/combined-pca-134.jpeg", width=10)
```


```{sh, eval=FALSE}
srun -p high -t 01:00:00 --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/106/134-lgs.beagle.gz -K 2 -o outputs/106/134-lgs.K2
srun -p high -t 01:00:00 --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/106/134-lgs.beagle.gz -K 3 -o outputs/106/134-lgs.K3 &
srun -p     high -t 01:00:00 --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/106/134-lgs.beagle.gz -K 4 -o outputs/106/134-lgs.K4 &
srun -p high -t 01:00:00 --nodes=1 $HOME/angsd/misc/NGSadmix -P 12 -likes outputs/106/134-lgs.beagle.gz -K 5 -o outputs/106/134-lgs.K5 &
```


