---
title: "107-downsampled-newseqs"
output: html_document
date: "2023-05-02"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
```

```{r}
meta<-read_csv("meta/meta260.csv")
meta$Location<-factor(meta$Location, levels=c("YBAK","SKNA","PTLC","HRLC","FRAS","LWSH",
                                              "NOOK","COLR",
                                              "HUMB","PETA","ALVS","SUIB","CHPI",
                                         "New Sequencing"))




m154<-meta %>% group_by(Location) %>% top_n(20, Dedup)
write_tsv(m154 %>% select(Path), file="bamlists/154.bamlist",col_names = FALSE)
write_tsv(m154 %>% select(Sample), file="bamlists/154.names",col_names = FALSE)
```


```{sh, eval=FALSE}

srun -p med -t 32:00:00 --mem=16G --nodes=1 $HOME/angsd/angsd -P 8 \
  -ref /home/maccamp/genomes/hypomesus-20210204/Hyp_tra_F_20210204.fa \
  -bam $HOME/longfin/bamlists/154.bamlist -GL 1 \
  -doGLF 2 -doMajorMinor 1 -doMaf 2 -SNP_pval 1e-6 -minMapQ 10 -minQ 20 -minMaf 0.05 \
  -rf  $HOME/longfin/outputs/106/lg01.txt  -minInd 123 \
  -out $HOME/longfin/outputs/107/154-lgs > outputs/107/beagle-lgs.out 2> outputs/107/beagle-lgs.err &

#
  
conda activate py2
srun -p high -t 01:00:00 python $HOME/pcangsd/pcangsd.py -beagle outputs/107/154-lgs.beagle.gz \
conda deactivate

srun -p med -t 14:00:00 --mem=16G --nodes=1 --ntasks=1 angsd -P 12 \
-bam bamlists/154.bamlist \
-out /home/maccamp/longfin/outputs/107/plink \
-anc /home/maccamp/genomes/hypomesus-20210204/Hyp_tra_F_20210204.fa \
-rf $HOME/delta-smelt/metadata/large-contigs.txt  \
-minInd 139 -minMaf 0.05  -minMapQ 10 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 4 -doPost 1 -postCutoff 0.95 -doPlink 2 >outputs/107/plink.out 2>outputs/107/plink.stderr &


```
