---
title: "108-existing-data"
output: html_document
date: "2024-12-15"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
```

```{r}
existing<-read_tsv("meta/counts.tsv") %>% filter(Alignment !="Original")
existing<-existing %>% filter(Alignment=="New Genome") %>%
  mutate(Path=paste0("/home/maccamp/data/longfin/",File)) %>% filter(Reads > 4e5)

meta<-existing %>% separate(Sample, into=c("Loc","Number"), sep="_", remove=FALSE)

#Factor N->S
meta$Location<-factor(meta$Loc, levels=c("YBAK","SKNA","PTLC","HRLC","FRAS","LWSH","COLR",
                                              "HUMB","SFBY","PETA","ALVS","SUIB","CHPI"))


ggplot(meta) + geom_histogram(aes(x=Reads, fill=Location)) +
  scale_fill_viridis_d()
```

Filter to SFE area samples

```{r}
sf<-meta %>% filter(Location %in% c("SFBY", "ALVS", "SUIB", "CHPI"))

ggplot(sf) + geom_histogram(aes(x=Reads, fill=Location)) +
  scale_fill_viridis_d(option="magma")

```

```{r}
write_csv(sf, "meta/sf44.csv")
sf %>% select(Path) %>% write_tsv("bamlists/sf44.bamlist", col_names = FALSE)
```

Lets quickly generate a covariance matrix.
```{sh,, eval=FALSE}
srun -p bigmemm -t 10:00:00 --mem=64G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12  $HOME/angsd/angsd -P 12 -bam bamlists/sf44.bamlist \
-minInd 33 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/108/sf44-ibs-75 >outputs/108/sf44-ibs-75.out 2> outputs/108/sf44-ibs-75.err &
```



```{r}
m <- as.matrix(read.table("outputs/108/sf44-ibs-75.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```


```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(sf)

pc12<-ggplot(covs) +
  geom_point(aes(x=V1, y=V2, shape=Location, fill=Reads), cex=3, alpha=0.75) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  scale_shape_manual(values=c(23,24,21,22)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  #scale_fill_viridis_c() +
  scale_fill_viridis_c(option="viridis") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("A. PCA of SFE samples from Saglam et al. n=44") +
  theme(plot.title = element_text(hjust=0, size=16)) #+
#  facet_wrap(.~clust_name_100)

pc12

ggsave("outputs/108/pca-saglam-sfe.jpeg")
```

```{r}
covs %>% filter(V1 > 0.4)
```


```{r}
covs %>% filter(V2 > 0.4)
```


Ack. Now to run again
```{r}
sf41<-sf %>% filter(!Sample %in% c("SFBY_075","SFBY_083","SFBY_081"))
sf41  %>% select(Path) %>% write_tsv("bamlists/sf41.bamlist", col_names = FALSE)
```

Lets quickly generate a covariance matrix.
```{sh,, eval=FALSE}
srun -p bigmemm -t 10:00:00 --mem=64G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12  $HOME/angsd/angsd -P 12 -bam bamlists/sf41.bamlist \
-minInd 31 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/108/sf41-ibs-75 >outputs/108/sf41-ibs-75.out 2> outputs/108/sf41-ibs-75.err &
```
